src/main/java/justin/TaskList.java

```diff
@@ -0,0 +1,174 @@
+            Todo td = new Todo(description);
+            list.add(td);
+            // formatting
+            printLineBreaker();
+            System.out.println("Got it. I've added this task:");
+            System.out.println(" " + td.toString());
+            System.out.println("Now you have " + list.size() + " tasks in the list");
+            printLineBreaker();
+        }
+
+    }
+
+    public void addEvent(String description, String date) {
+
+        boolean ifExist = false;
+
+        for (int i = 0; i < list.size(); i++) {
+            if(list.get(i).description.equals(description)) {
+                Event e = new Event(description, date);
+                // bringing over info from superclass
+                if (list.get(i).isDone) {
+                    e.markAsDone();
+                }
+
+                list.set(i, e); // insert into list
+
+                //formatting
+                printLineBreaker();
+                System.out.println("Got it. I've added this task:");
+                System.out.println(" " + e.toString());
+                System.out.println("Now you have " + list.size() + " tasks in the list");
+                printLineBreaker();
+                ifExist = true;
+            }
+        }
+
+        if (!ifExist) {
+            Event e = new Event(description, date);
+            list.add(e);
+            //formatting
+            printLineBreaker();
+            System.out.println("Got it. I've added this task:");
+            System.out.println(" " + e.toString());
+            System.out.println("Now you have " + list.size() + " tasks in the list");
+            printLineBreaker();
+        }
+    }
+
+    public void delete(String num) throws JustinException {
+        try {
+
+            int listNum = Integer.parseInt(num); // changes to int
+            Task newTask = list.remove(listNum-1); // delete the entry of choice
+            //format
+            printLineBreaker();
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(" " + newTask.toString());
+            System.out.println("Now you have " + list.size() + " tasks in the list");
+            printLineBreaker();
+
+        } catch(IndexOutOfBoundsException e) {
+            throw new JustinException("OOPS!!! Cannot delete what you don't have!");
+        }
+    }
+
+    public ArrayList<String> find(String key) {
+
+        boolean isFound = false; // if any tasks matches description we can return a justin.TaskList
+        ArrayList<String> holder = new ArrayList<>();
+
+        for (int i = 0; i < list.size(); i++) {
+
+            String description = list.get(i).getDescription();
+
+            if (description.contains(key)) {
+                holder.add(list.get(i).toString());
+            }
+
+            isFound = true; // there is at least ONE instance of description with key word
+        }
+
+
+        return holder;
+
+    }
+
+
```

Comment:
I think there are times when it is not really necessary to have multiple blank lines in between parts of your code?